frontend/src/App.tsx

```diff
@@ -1,40 +1,40 @@
 import "./assets/App.scss";
 
 import axios from "axios";
-import React, { useEffect, useRef, useState } from "react";
+import * as React from "react";
+import { ActionMeta, OptionsType } from "react-select";
 
 import Chart from "./components/Chart";
 import Sidebar from "./components/Sidebar";
+import { Campaign, Data, SelectOption } from "./types/types";
 
-const App = () => {
-  const baseDataUrl = "http://localhost:8000/api/data/";
-  const baseCampaignUrl = "http://localhost:8000/api/campaign/";
+const baseDataUrl = "http://localhost:8000/api/data/";
+const baseCampaignUrl = "http://localhost:8000/api/campaign/";
 
-  const [data, setData] = useState([]);
-  const [campaignsList, setCampaignsList] = useState([]);
-  const [dataSources, setDataSources] = useState([]);
-  const [campaigns, setCampaigns] = useState([]);
-  const [pageSize, setPageSize] = useState(10);
-  const [errorMsg, setErrorMsg] = useState("");
-  const [dataUrl, setDataUrl] = useState(baseDataUrl);
+const App = (): React.ReactElement => {
+  const [data, setData] = React.useState<ReadonlyArray<Data>>([]);
+  const [campaignsList, setCampaignsList] = React.useState<
+    ReadonlyArray<Campaign>
+  >([]);
+  const [dataSources, setDataSources] = React.useState<string[]>([]);
+  const [campaigns, setCampaigns] = React.useState<string[]>([]);
+  const [pageSize, setPageSize] = React.useState(10);
+  const [errorMsg, setErrorMsg] = React.useState("");
+  const [dataUrl, setDataUrl] = React.useState(baseDataUrl);
 
-  const latestBaseDataUrl = useRef(dataUrl);
+  const latestBaseDataUrl = React.useRef(dataUrl);
 
   /**
    * Function for detecting url change.
-   * @param {string} prevUrl - prevUrl
-   * @param {string} nextUrl - nextUrl
-   * @returns {boolean} - true if urls are different
    */
-  const areUrlsDifferent = (prevUrl, nextUrl) => prevUrl !== nextUrl;
+  const areUrlsDifferent = (prevUrl: string, nextUrl: string) =>
+    prevUrl !== nextUrl;
 
   /**
-   * Function for getting data from the API.
-   * @param {string} url - api endpoint url
-   * @returns {Promise} - returns axios promise
+   * Function for getting data from API.
    */
-  const getData = async (url) => {
-    const response = await axios.get(url);
+  const getData = async <T extends Record<string, unknown>>(url: string) => {
```

Comment:
Could go into some data fetching wrapper to abstract axios away from the component